### PR TITLE
Set application ids in enrollments table

### PR DESCRIPTION
**Pull request by megcrenshaw**
I used an adhoc (finally) to ensure this method worked to set application ids correctly in the enrollments table.

Originally (see https://github.com/code-dot-org/code-dot-org/pull/44205), I called `.save!` in enrollments to trigger `set_application_id`. This approach threw `Validation failed: Email already enrolled in workshop (ActiveRecord::RecordInvalid)`, which stopped the process. I could modify the script to continue in cases where the validation failed, but we want to update the application id even if the email is already enrolled in a workshop (we don't care––they should still have their application id linked to their enrollment). Happy to chat more if this seems bad.

So, I switched to calling `.set_application_id` directly. Originally, this approach threw `from dashboard/app/models/pd/enrollment.rb:336:in 'set_application_id': undefined method 'school_year' for nil:NilClass (NoMethodError)`. I modified the `enrollments.rb` to instead use `&.` for that. That approach should also fix the errors popping up when we need to purge accounts (see [discussion](https://codedotorg.slack.com/archives/C55JZ1BPZ/p1642897640004000)).

When run on an adhoc, the script displayed at the end, `Finished updating enrollments: set 41759 application ids`.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/pd/enrollment.rb [hunk 1]
```diff
@@ -329,13 +329,13 @@ def newly_accepted_facilitator?
...
-    course_match = ->(application) {Pd::Application::ApplicationBase::COURSE_NAME_MAP.dig(application.try(:course).to_sym) == workshop.try(:course)}
+    course_match = ->(application) {Pd::Application::ApplicationBase::COURSE_NAME_MAP.dig(application.try(:course)&.to_sym) == workshop.try(:course)}
     pd_match = ->(application) {application.try(:pd_workshop_id) == pd_workshop_id}
 
     application_id = nil
     # Finds application from the school year of the workshop. Assumes workshops start after 6/1
     # because workshop.school_year assumes 6/1 is the start of the school year
-    Pd::Application::ApplicationBase.where(user_id: user_id, application_year: workshop.school_year).each do |application|
+    Pd::Application::ApplicationBase.where(user_id: user_id, application_year: workshop&.school_year).each do |application|
       application_id = application.id if course_match.call(application) || pd_match.call(application)
```
Comment (bethanyaconnor) [inline dashboard/app/models/pd/enrollment.rb:339]: Can `course_match` or `pd_match` ever be nil?

Comment (megcrenshaw) [reply dashboard/app/models/pd/enrollment.rb:339]: No––they're defined above (lines 332 and 333) and return true or false. Should we be doing `&.` here as well to be super safe?

Comment (megcrenshaw) [reply dashboard/app/models/pd/enrollment.rb:339]: So I suppose if we delete course_match or pd_match in that function, we're toast

Comment (bethanyaconnor) [reply dashboard/app/models/pd/enrollment.rb:339]: ahh right they're procs. 🤦‍♀️  Sorry, no need to do anything with them, got caught up in using `&` haha

**Review by bethanyaconnor — APPROVED** [review]

File: dashboard/app/models/pd/enrollment.rb (hunk 1 above)
Comment (megcrenshaw) [reply dashboard/app/models/pd/enrollment.rb:339]: Yeah, makes sense. Thanks for asking!

**Pull request merged**